websites/F/Facepunch Wiki/presence.ts

```diff
@@ -0,0 +1,28 @@
+const presence = new Presence({
+  clientId: "782958551123361822"
+});
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "icon"
+  };
+
+  function setTimestamp(): number {
+    return Math.floor(Date.now() / 1000);
+  }
+
+  if (window.location.pathname === "/") {
+    presenceData.details = await presence.getSetting("homepageMessage");
+    presenceData.startTimestamp = setTimestamp();
+  } else {
+    presenceData.details = await presence.getSetting("viewingMessage") + " " + document.title;
+    presenceData.startTimestamp = setTimestamp();
+  }
```

Comment:
maybe add some more statuses?
and I would personally put the document.title then on presenceData.state